autoload/ale/floating_preview.vim

```diff
@@ -106,26 +106,39 @@ function! s:NvimPrepareWindowContent(lines) abort
     let l:width += 2
     let l:height += 2
 
-    let l:hor          = g:ale_floating_window_border[0]
-    let l:top          = g:ale_floating_window_border[1]
-    let l:top_left     = g:ale_floating_window_border[2]
-    let l:top_right    = g:ale_floating_window_border[3]
-    let l:bottom_right = g:ale_floating_window_border[4]
-    let l:bottom_left  = g:ale_floating_window_border[5]
+    if len(g:ale_floating_window_border) == 6
+        let l:left         = g:ale_floating_window_border[0]
```

Comment:
I would add the new boder options to the end of the array so the first 6 entries stay compatible without any change and then modify this code to use safer array indexing:

```
let l:left                      = get(g:ale_floating_window_border, 0, '|')
let l:top                      = get(g:ale_floating_window_border, 1, '-')
let l:top_left               = get(g:ale_floating_window_border, 2, '+')
let l:top_right             = get(g:ale_floating_window_border, 3, '+')
let l:bottom_left         = get(g:ale_floating_window_border, 4, '+')
let l:bottom_right       = get(g:ale_floating_window_border, 5, '+')
let l:right                    = get(g:ale_floating_window_border, 5, l:left)
let l:bottom                = get(g:ale_floating_window_border, 5, l:bottom)
```

This not only simplies logic it also makes the code safer in case someone sets the g:ale_floating_window_border array wrong.